### PR TITLE
Ngeo import

### PR DESCRIPTION
**Pull request by oterral**
[test](https://mf-geoadmin3.int.bgdi.ch/teo_ngeo2/index.html?topic=ech&lang=fr&bgLayer=ch.swisstopo.pixelkarte-farbe&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bav.haltestellen-oev,ch.swisstopo.swisstlm3d-wanderwege&layers_visibility=false,false,false,false&layers_timestamp=18641231,,,)

To do:

- [x] try to use css file from ngeo
- [x] try to use the ngeo.File service in all the application if possible
- [x] fix translations of ngeo component -> wait for https://github.com/camptocamp/ngeo/pull/2185
- [x] find why the coverage.txt file is disappeared
- [x] fix jenkins
- [x] verfiy that all Makefile targets work
- [x] apply th patch for #3547 in ngeo if necessary
- [x] fix button to open suggestions in ngeo-> wait for https://github.com/camptocamp/ngeo/pull/2186
- [x] fix e2e tests
- [x]  ~~fix width if possible see [comment](https://github.com/geoadmin/mf-geoadmin3/pull/3596#issuecomment-258599059)~~ -> impossible without creating bugs
- [x] finish translation see [comment](https://github.com/geoadmin/mf-geoadmin3/pull/3596#issuecomment-258599059)
- [x]  fix dev mode on Firefox and IE=11 see [comment](https://github.com/camptocamp/ngeo/pull/2158#discussion_r94959491)
- [x]  avoid copy of all ngeo files when we deploy 

IE <=10 in dev mode doesn't work

### CODE REVIEW

**Comment by davidoesch** [comment]
we have to state more prominent that you can not only import but WHAT you can import

Proposal: we add more specific terms

image

**Comment by cedricmoullet** [comment]
If you hover the copyrights links: 
image
Then you don't see the tooltip (it's behind the background manager)

https://mf-geoadmin3.int.bgdi.ch/teo_ngeo2/index.html?topic=ech&lang=fr&bgLayer=ch.swisstopo.pixelkarte-farbe&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bav.haltestellen-oev,ch.swisstopo.swisstlm3d-wanderwege,WMS%7C%7COrthophoto%20Yverdon%20Y-Parc%7C%7Chttp:%2F%2Fogc.heig-vd.ch%2Fmapserver%2Fwms%3F%7C%7Cr-pod_yverdon_y-parc%7C%7C1.3.0&layers_visibility=false,false,false,false,true&layers_timestamp=18641231,,,,&X=179611.72&Y=539127.69&zoom=13

**Comment by gberaudo** [comment]
@oterral, I have started integrating the import in ngeo.

It looks like the version of typeahead used in geoadminis pretty old (0.9.3 Juine 26th 2013). Is there a reason for sticking with this version?

It is blocking me since the geoadmin code makes use of functionalities not present anymore in recent versions of typeahead:
- $(taElt).data('ttView');
- dataset.getSuggestions;
- ... probably longer list.

Could you migrate geoadmin code (including the import code from this PR)?

In ngeo we recently switched to https://github.com/corejavascript/typeahead.js which is a maintained version of typeahead (last release a few weeks ago). It would be great if you used it as well.

**Comment by oterral** [comment]
Yes we stick to this version because the migration to a new version was not obvious. It worked and we don't use typeahead anywhere else.  

I can give it a try to replace the initSuggestion function.

 Which version do you used ?  the typeahead.jquery.js version?

**Comment by gberaudo** [comment]
Thanks @oterral. We are using `typeahead.bundle.js` version 1.0.1.

**Comment by oterral** [comment]
@gberaudo typeahead updated.

**Comment by gberaudo** [comment]
Thanks @oterral, I will integrate your changes in the ongoing https://github.com/camptocamp/ngeo/pull/2070 PR.

**Comment by oterral** [comment]
jenkins test

**Comment by davidoesch** [comment]
See my comment from 5.11

**Comment by oterral** [comment]
@davidoesch feel free to change translations ont the google doc . The last 3 lines.

**Comment by davidoesch** [comment]
updated import_online_placeholder,import_tooltip

**Comment by oterral** [comment]
ready for review

**Comment by gjn** [comment]
Rebase and I'll review and merge.

**Comment by oterral** [comment]
@gjn rebased

File: scripts/s3manage.py [hunk 1]
```diff
@@ -225,8 +225,16 @@ def upload(bucket_name, base_dir, deploy_target, named_branch):
...
                   'checker', 'geoadmin.%s.appcache' % version)
 
+    # exclude some ngeo files
+    ngeo_prefix = '/ngeo'
+    include_ngeo_folders = ['src/modules/import']
+
     for directory in upload_directories:
         for file_path_list in os.walk(os.path.join(base_dir, directory)):
+            file_path = file_path_list[0]
+            print file_path
```
Comment (loicgasser) [inline scripts/s3manage.py]: you want to leave that?

File: scripts/clonebuild.sh [hunk 2]
```diff
@@ -46,6 +47,8 @@ else
   update_and_reset_git_project
 fi
 
+git submodule update --init
```
Comment (loicgasser) [inline scripts/clonebuild.sh]: you will need to update deploydev.sh as well

File: scripts/s3manage.py [hunk 3]
```diff
@@ -225,8 +225,16 @@ def upload(bucket_name, base_dir, deploy_target, named_branch):
                   'robots.txt', 'robots_prod.txt', 'favicon.ico',
                   'checker', 'geoadmin.%s.appcache' % version)
 
+    # exclude some ngeo files
+    ngeo_prefix = '/ngeo'
+    include_ngeo_folders = ['src/modules/import']
```
Comment (loicgasser) [inline scripts/s3manage.py]: exclude?

File: scripts/s3manage.py [hunk 4]
```diff
@@ -225,8 +225,16 @@ def upload(bucket_name, base_dir, deploy_target, named_branch):
...
+
     for directory in upload_directories:
         for file_path_list in os.walk(os.path.join(base_dir, directory)):
+            file_path = file_path_list[0]
```
Comment (loicgasser) [inline scripts/s3manage.py]: same thing is done later line 241, so I would make this check after 239

File: src/components/gettext/GetTextModule.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+goog.provide('gettext');
+
+(function() {
+
+  /**
+   * This module is only use to provide translations to ngeo module.
```
Comment (loicgasser) [inline src/components/gettext/GetTextModule.js]: used

**Comment by loicgasser** [comment]
I found a small issue with a translation in the menu.
image

**Comment by loicgasser** [comment]
trying to install it in my env results in a failure

after

```make cleanall user
```
I get
```
FileError: '../ngeo/src/modules/import/style/importonline.less' wasn't found. Tried - /home/ltgal/mf-geoadmin3/src/ngeo/src/modules/import/style/importonline.less,/home/ltgal/mf-geoadmin3/src/ngeo/src/modules/import/style/importonline.less,../ngeo/src/modules/import/style/importonline.less in /home/ltgal/mf-geoadmin3/src/style/app.less on line 17, column 1:
16 @import "../components/topic/style/topic.less";
17 @import "../ngeo/src/modules/import/style/importonline.less";
18 @import "../ngeo/src/modules/import/style/importlocal.less";

Makefile:556: recipe for target 'src/style/app.css' failed
make[1]: *** [src/style/app.css] Error 1
make[1]: Leaving directory '/home/ltgal/mf-geoadmin3'
Makefile:162: recipe for target 'user' failed
make: *** [user] Error 2
```
Shouldn't git submodule init and git submodule foreach git .... ?

File: scripts/s3manage.py (hunk 3 above)
Comment (oterral) [inline scripts/s3manage.py]: exclude what?

Comment (oterral) [inline scripts/s3manage.py]: you talk about the comment?

**Comment by oterral** [comment]
You need to do it manually it's in the doc.

**Comment by gjn** [comment]
A build from sratch should always be sucessfull.

File: scripts/clonebuild.sh (hunk 2 above)
Comment (oterral) [inline scripts/clonebuild.sh]: yes probably

**Comment by loicgasser** [comment]
I would suggest a ngeodev target instead.
Main cmd should be straightforward

File: src/components/gettext/GetTextModule.js (hunk 5 above)
Comment (oterral) [inline src/components/gettext/GetTextModule.js]: fixed

File: scripts/s3manage.py (hunk 4 above)
Comment (oterral) [inline scripts/s3manage.py]: yes fixed

File: scripts/s3manage.py (hunk 3 above)
Comment (oterral) [inline scripts/s3manage.py]: fixed

File: scripts/clonebuild.sh (hunk 2 above)
Comment (oterral) [inline scripts/clonebuild.sh]: yes fixed

**Comment by oterral** [comment]
It is straigthforward if you have follow the doc ;)

```
git clone git@github.com:geoadmin/mf-geoadmin3.git
git submodules update --init
make cleanall user
```

When you develop in a submodule there is nothing worse than a build which try to update/override/init stuff that you don't want.

**Comment by oterral** [comment]
I can check if the module is initialized yes.

**Comment by loicgasser** [comment]
yes but that's not necessary true...
git submodules update --init

creates ngeo folder in `src/ngeo`

check if it exists, and in cleanall remove it

**Comment by loicgasser** [comment]
you can then remove submodule commands in deployscripts

**Comment by oterral** [comment]
I've a problem I need to have ngeo files before runing any make command.
See the SRC_ES6_FILES variables at the beginning.

**Comment by gjn** [comment]
Remind me: why can't we treat it like other external libraries? (like ol3 or ol3-ceiusm)

**Comment by gjn** [comment]
```
I've a problem I need to have ngeo files before runing any make command.
See the SRC_ES6_FILES variables at the beginning.
```
This could be handled by a standard pre-requisite, no? If not exist, do something.

**Comment by oterral** [comment]
we must compile the ngeo files with our build to avoid having all the library loaded and to take advantage of the goog.require to load only what we need

**Comment by gjn** [comment]
Hmm. there must be another way. Could be part of the SRC_JS_FILES line, or some pre-requisite definition that is smart and a target that uupdates the SRC_JS_FILES list.

File: Makefile [hunk 6]
```diff
@@ -378,7 +381,7 @@ prd/lib/build.js: src/lib/polyfill.min.js \
 	    src/lib/slip.min.js \
 	    src/lib/bootstrap.min.js \
 	    src/lib/moment-with-customlocales.min.js \
-	    src/lib/typeahead-0.9.3.min.js \
+	    src/lib/typeahead.jquery.min.js \
```
Comment (gjn) [inline Makefile]: Why this change?

**Comment by gjn** [comment]
From what I understand, we only ever use code under ngeo`ngeo/src/modules/import`. Why not treat this code like ol3-cesium.

1) We have a target to update it (based on commit hash of git repo) in src/ngeo/modules/import
2) We keep track of these files directly in mf-geoadmin3 (we have ol3-cesium debug ond compiled tracked as well)

Thougths?

File: Makefile (hunk 6 above)
Comment (loicgasser) [inline Makefile]: Same version name as ngeo I presume

**Comment by loicgasser** [comment]
I think it s fine. We treat ngeo the same way we treat our components but we commit it in ngeo. I don t see any issue with it as long as build is straightforward.

We can have a target to update it. Actually it would be a good idea to keep the sha like we do for olcesium. 

But updating it is quite easy: git submodule foreach git pull origin master 
Or 
git submodule foreach git fetch && git submodule foreach reset --hard {sha}

**Comment by oterral** [comment]
@gjn we can't provide an external ngeo file with only ngeo.

We could compile ngeo with ol3-cesium and ol3  but you will have all the ngeo library and you couldn't use  services like ngeoFIleService because it will be renamed.

With this solution you load only what you need from ngeo and it's easier to test new functionnality of the library.

File: Makefile (hunk 6 above)
Comment (oterral) [inline Makefile]: Yes ngeo use the last version of typeahaead.

**Comment by oterral** [comment]
We found an ugly way to init module on each use of `make` command, [see](https://github.com/geoadmin/mf-geoadmin3/pull/3596/files#diff-b67911656ef5d18c4ae36cb6741b7965R4).

I'va also added a `make ngeo` target to update the submodule with a NGEO_VERSION variable like ol3cesium.

**Comment by loicgasser** [comment]
cool, tooltip translation is still not ok though otherwise LGTM.

**Comment by oterral** [comment]
It's weird for the tooltip I have it for all translations.

**Comment by loicgasser** [comment]
`rm -rf src/ngeo` in cleanall ?

**Comment by oterral** [comment]
No ngeo folder is added in github and you need it to set the submodule.

**Comment by gjn** [comment]
Ha! Thanks alot for making the effort. I'll review tomorrow.

File: src/locales/de.json [hunk 7]
```diff
@@ -2,14 +2,33 @@
 "3d_ie11_alert": "Sie verwenden InternetExplorer <= 11: Die 3D Darstellung wird nicht performant sein. Bitte nutzen Sie die neueste Version von Microsoft EDGE, Firefox oder Chrome um beste Ergebnisse zu erhalten",
 "3d_overlay_warning": "Wichtiger Hinweis: 2D-Daten in 3D.<br/>Sie betrachten jetzt Daten, die für die Darstellung in 3D nicht geeignet sind. Dies gilt insbesondere für Karten und Luftbilder. Diese werden in steilem Gelände bei der Nutzung in 3D verzerrt dargestellt\n",
 "3d_render_error": "Systemvorraussetzung für 3D Ansicht nicht erfüllt. Damit die 3D-Darstellung funktioniert, muss Ihr Gerät folgende Anforderungen erfüllen.Nutzen Sie die neueste Version von Microsoft EDGE, Firefox oder Chrome. Verwendung einer aktuellen Grafikkarte mit aktuellen Treibern. .",
+"Add layer": "Layer hinzufügen",
```
Comment (gjn) [inline src/locales/de.json]: why do we suddenly change naming conventions of translation tokens? I don't like to mix these...

Also, they are not in empty.json file - or under the old convention.

Comment (gjn) [inline src/locales/de.json]: As I understand, it's because ngeo does it like that?

The we should move them to the end of the files. And assure our translations really work. Don't we need them in empty.json? Or does ngeo take care of the translations?

Comment (oterral) [inline src/locales/de.json]: Yes it caused by ngeo.

Don't forget de.json is auto generated.
In empty.json file (in the excel file too) we keep the same naming conventions:
```
"add_layer": "Add layer",
```
The text "Add layer" represents the corresponding message id used by ngeo.

Comment (gjn) [inline src/locales/de.json]: hmm, but `Add layer` is value in empty, but key in translations. Strange, no?

Comment (oterral) [inline src/locales/de.json]: It's necessary to avoid to change of the naming convention in the excel file.

**Comment by gjn** [comment]
@davidoesch Can we go live with this next Wednesday?

I'll make sure that the help is updated by then.

**Comment by gjn** [comment]
Thanks alot for this. We'll go live with this next week if we are able to prepare the help until then and @davidoesch agrees.

**Comment by gjn** [comment]
@oterral @davidoesch Please validate the texts, then we go live.

Thanks for adapting the build. Good stuff.

**Comment by davidoesch** [comment]
See https://github.com/geoadmin/mf-geoadmin3/pull/3596#issuecomment-258599059 

"Downloading file "  for WMS does not make sense.. Happy to change translations

-> 'll need in depth testing ( and then adapt the help as well)

**Comment by gjn** [comment]
Help will be done by Paulo.

The function has beend well tested. You need more time for testing?

@oterral Can you adapt the text according to Dave's proposal?

**Comment by davidoesch** [comment]
@mariokeusen will test

**Comment by oterral** [comment]
@davidoesch where do you see "Downloading file" ?

**Comment by oterral** [comment]
The textes to translate are in yellow on the google doc

**Comment by davidoesch** [comment]
[ ]  "Downloading!"
- Select a slow wms in pulldown 
- click connect 
- then field turns red and says downloading 
-> there is no such msgid. change it to "loading.." (since we don't download the data) you can even use for ONLINE and LOKAL msgid parse_succeeded and parsing_file

[ ] add the text in IT 
image

as well for DE FR EN ( it does not appear like this in this lng right now)


I changed the follwowing ID's (since parsing will not be understood by my mother)

[ ] parse_succeeded
[ ] parsing_file

**Comment by mariokeusen** [comment]
Tested a bit as well and encountered following issues: 

- Local kml's: does not work. It looks like there is a timeout while loading (yes, I've used a valid kml file)
- The online way works well, but I cannot submit an online embedded kml-file with a permalink. The kml disappears by opening the permalink. On the other hand it works with wms sources.
- (nice to have, not directly bound on this ticket) the old wish for a "zoom to layer extent" icon, if you click this PL and search for the data. The wms-source only covers a very tiny part of the mapviewer's b-box, and therefore it's quite hard to find something. https://mf-chsdi3.int.bgdi.ch/shorten/72397804a8

**Comment by mariokeusen** [comment]
Addition: local kml's. Actually does work, but the button's translation is confusing:
image

**Comment by gjn** [comment]
@oterral Please address the issues.

**Comment by oterral** [comment]
@davidoesch @mariokeusen 

Hope it's good this time: [Test](https://mf-geoadmin3.int.bgdi.ch/teo_ngeo2/index.html)

**Comment by oterral** [comment]
Test link failed, deploy on int is broken

**Comment by oterral** [comment]
[Test](https://mf-geoadmin3.dev.bgdi.ch/ltteo/?lang=fr&topic=ech&bgLayer=ch.swisstopo.pixelkarte-farbe&X=190000.00&Y=660000.00&zoom=1&dev3d=true&debug&layers=ch.swisstopo.zeitreihen,ch.bfs.gebaeude_wohnungs_register,ch.bav.haltestellen-oev,ch.swisstopo.swisstlm3d-wanderwege&layers_visibility=false,false,false,false&layers_timestamp=18641231,,,)

**Comment by davidoesch** [comment]
tested on latest chrome ff IE edge on win10
ok

**Comment by gjn** [comment]
Which goes first. OGCProxy? Or This? @oterral @loicgasser

**Comment by loicgasser** [comment]
I guess this has been out there for a while and we re still waiting for big 3 approval.

**Comment by davidoesch** [comment]
Ping @mariokeusen @cedricmoullet

**Comment by oterral** [comment]
Deploybranch fixed

**Comment by gjn** [comment]
@davidoesch @cedricmoullet @pauloamado We will deploy this tomorrow.

**Comment by davidoesch** [comment]
@pauloamado can you update the help https://help.geo.admin.ch/?id=62 and https://help.geo.admin.ch/?id=63 after the deploy? thanks

**Comment by gjn** [comment]
Thanks alot everybody for the efforts and the patience.
